examples/token/main.aleo

```diff
@@ -17,6 +21,76 @@ record token:
     // The token amount.
     amount as u64.private;
 
+/* Approve */
+
+function approve_public:
+    input r0 as address.public; // approver
+    input r1 as address.public; // spender
+    input r2 as u64.public; // amount spender is allowed to withdraw from approver
+
+    finalize r0 r1 r2;
+
+finalize approve_public:
+    input r0 as address.public; // approver
+    input r1 as address.public; // spender
+    input r2 as u64.public; // increase in amount spender is allowed to withdraw from approver
+
+    hash.ped64 r0 into r3; // hash approver into field
+    hash.ped64 r1 into r4; // hash spender into field
+
+    mul r3 r4 into r5; // multiply approver f by spender f
+
+    // if approvals for approver field times spender field exists, the approved amount is increased.
+    // otherwise, the approved allowance is created.
+    increment approvals[r5] by r2;
+
+/* Unapprove */
+
+function unapprove_public:
+    input r0 as address.public; // approver
+    input r1 as address.public; // spender
+    input r2 as u64.public; // amount spender's allowance is decreasing by
+
+    finalize r0 r1 r2;
+
+finalize unapprove_public:
+    input r0 as address.public; // approver
+    input r1 as address.public; // spender
+    input r2 as u64.public; // decrease in amount spender is allowed to withdraw from approver
+
+    hash.ped64 r0 into r3; // hash approver into field
+    hash.ped64 r1 into r4; // hash spender into field
+
+    mul r3 r4 into r5; // multiply approver field by spender field
+
+    // if approvals for approver field times spender field exists, the approved amount is decreased.
+    // otherwise, the approved allowance is created.
+    decrement approvals[r5] by r2;
```

Comment:
Can we access the value of a mapping? It would be nice if the decrease to the approvals didn't underflow and instead was reset to 0 in the case that we're trying to decrease it by too much.

We tried:
```
gt approvals[r5] r2 into r6;
ternary r6 r2 approvals[r5] into r7;
decrement approvals[r5] by r7;
```